### PR TITLE
precommit: switch `flake8` with Ruff

### PR DESCRIPTION
**Pull request by Borda**
**Just a suggestion and show what it would mean...**

Recently [Ruff](https://github.com/astral-sh/ruff) has become very popular among also mainstream projects for:
- much faster, they say 70x
- integrate several other tools under one roof, such as isort or bandit

### CODE REVIEW

File: .pylintrc
Comment (Borda) [inline .pylintrc:1]: this was moved to the `pyproject.toml`

File: pytest.ini
Comment (Borda) [inline pytest.ini:1]: this was moved to the `pyproject.toml`

File: setup.cfg [hunk 1]
```diff
@@ -1,13 +1,3 @@
-# [bumpversion]
-# current_version = 0.1.18
-
-[bdist_wheel]
-# Use this option if your package is pure-python
-universal = 1
-
-[metadata]
```
Comment (Borda) [inline setup.cfg:8]: this was moved to the `pyproject.toml`

**Comment by shaypal5** [comment]
Ok, a few questions:

1. What's the "W" linter?
E is pycodestyle and F is pyflakes (at least in the Ruff tutorial).

2. Can you gauge the effect of running it with the suggested configuration?

3. What not start just with one of the flake8 implementations (I see flake8-bugbear, denoted with "B", is popular enough) for this PR, and then suggest incremental additions of a single additional linter at a time? This will make a discussion on the topic much easier.

I'll approve a PR moving our flake8 linting to Ruff without much discussion. Let's start with that, please.
@Borda

**Comment by Borda** [comment]
> 1. What's the "W" linter? E is pycodestyle and F is pyflakes (at least in the Ruff tutorial).

They are `pycodestyle` warnings - https://docs.astral.sh/ruff/rules/#warning-w

> 2. Can you gauge the effect of running it with the suggested configuration?

This is already applied, so as we did not add any new rules, just 1:1 with `flake8` the result is identical

> 3. What not start just with one of the flake8 implementations (I see flake8-bugbear, denoted with "B", is popular enough) for this PR, and then suggest incremental additions of a single additional linter at a time? This will make a discussion on the topic much easier.

yes exactly the is just rewrite `flake8` without any addition and if any other full shall be enabled, lest do each in separate PR

**Review by shaypal5 — APPROVED** [review]

**Pull request merged**